doc/source/ray-observability/state/state-api.rst

```diff
@@ -0,0 +1,303 @@
+
+E.g. Summarize all actors (e.g. number of alive actors, different actor classes, etc)
+
+.. code-block:: bash
+
+    ray summary actors
+
+E.g. Summarize all tasks (e.g. task count in different states, type of different tasks, etc)  
+
+.. code-block:: bash
+
+    ray summary tasks
+
+E.g. Summarize all objects (e.g. total number of objects, size of all objects, etc) 
+
+.. code-block:: bash
+
+    # To get callsite info, set env variable `RAY_record_ref_creation_sites=1` when starting the ray cluster
+    # RAY_record_ref_creation_sites=1 ray start --head
+    ray summary objects 
+
+List
+----
+
+Get a list of resources, possible resources include: 
+
+:ref:`Actors <actor-guide>`
+:ref:`Tasks <ray-remote-functions>`
+:ref:`Objects <objects-in-ray>`
+:ref:`Jobs <jobs-overview>`
+:ref:`Placement Groups <ray-placement-group-doc-ref>`
+:ref:`Nodes (Ray worker nodes)`
+:ref:`Workers (Ray worker processes)`
+:ref:`Runtime environments <runtime-environments>`
+
+E.g. List all nodes
+
+.. code-block:: bash
+
+    ray list nodes 
+
+E.g. List all placement groups
+
+.. code-block:: bash
+
+    ray list placement-groups
+
+Listing resources with one or multiple filters 
+List results could also be filtered. For example: 
+E.g. List local referenced objects created by a process
+
+.. code-block:: bash
+
+    ray list objects -f pid=12345 -f reference_type=LOCAL_REFERENCE
+
+E.g. List alive actors
+
+.. code-block:: bash
+
+    ray list actors -f state=ALIVE
+
+E.g. List running tasks
+
+.. code-block:: bash
+
+    ray list tasks -f scheduling_state=RUNNING
+
+E.g. List non-running tasks
+
+.. code-block:: bash
+
+    ray list tasks -f shceduling_state!=RUNNING
+
+Get
+---
+
+E.g. Get a task info
+
+.. code-block:: bash
+
+    ray get tasks <worker_id> 
+
+E.g. Get a node info
+
+.. code-block:: bash
+
+    ray get nodes <node_id> 
+
+
+Logs
+----
+
+State API also allows you to conveniently access ray logs. Note that you cannot access the logs from a dead node.
+By default, the API prints log from a head node.
+
+E.g. Get all retrievable log file names
+
+.. code-block:: bash
+
+    ray logs 
+
+E.g. Get a particular log file from a node
+
+.. code-block:: bash
+
+    # You could get the node id / node ip from `ray list nodes` 
+    ray logs gcs_server.out --node-id <XYZ> 
+
+E.g. Stream a log file from a node
+
+.. code-block:: bash
+
+    ray logs -f raylet.out --node-ip 172.31.47.143
+
+E.g. Stream actor log with actor id 
+
+.. code-block:: bash
+
+    # You could use ray list actors to get the actor ids
+    ray logs --actor-id=<XXX>
+
+E.g. Stream log from a pid 
+
+.. code-block:: bash
+
+    ray logs --pid=<XXX> --follow
+
+Failure Semantics
+-----------------
+
+The state APIs don't guarantee to return the correct snapshot of the cluster all the time. By default,
+all Python SDKs raise an exception when there's a missing output from the API. And CLI returns a partial result
```

Comment:
Oh right - we need to talk about Python here also I think. Maybe throw in some examples. 

Or maybe we could do this for all the examples (CLI and Python vs SSH and HTTPS) : 
<img width="795" alt="image" src="https://user-images.githubusercontent.com/11676094/180500371-0608bc17-d5da-4b34-8edb-fe00ea9da60f.png">